drivers/hmis/app/graphql/mutations/restore_scan_card_code.rb

```diff
@@ -0,0 +1,26 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class RestoreScanCardCode < CleanBaseMutation
+    argument :id, ID, required: true
+
+    field :scan_card_code, Types::HmisSchema::ScanCardCode, null: true
+
+    def resolve(id:)
+      code = Hmis::ScanCardCode.with_deleted.find(id)
+      raise 'unauthorized' unless current_permission?(permission: :can_manage_scan_cards, entity: code.client)
+
+      return code unless code.deleted_at.present?
```

Comment:
I don't think returning code is what we want, it's not the right shape.

Suggest we raise an error instead:
```suggestion
      raise unless code.deleted?
      # or
      return { scan_card_code: code } unless code.deleted?
```
Or even better, require the code to be deleted
```ruby
      code = Hmis::ScanCardCode.only_deleted.find(id)
```

Reply:
for some reason I was intending the mutations to be lenient, not error if trying to delete something thats already deleted. In case there is wonkiness in the UI where that happens. But it doesn't match patterns in other mutations, which are stricter on input. so I can change it back to that